pygfx/utils/text/_shaper.py

```diff
@@ -41,29 +41,34 @@ def shape_text(text, font_filename, direction=None):
 class TemporalCache:
     """A simple cache that drops old items based on time."""
 
-    def __init__(self, lifetime):
+    def __init__(self, lifetime, *, getter, minimum_items=0):
         self._ref_lifetime = lifetime
+        self._minimum_items = minimum_items
         self._cache = {}
         self._lifetimes = {}
+        self._getter = getter
 
-    def get(self, key):
-        """Gets the object corresponding to the given key, or None.
-        A successful get resets the item's lifetime.
+    def __getitem__(self, key):
+        """Gets the object corresponding to the given key.
+
+        Will reset the item's lifetime.
         Getting triggers the lifetimes of all items to be checked.
         """
-        try:
+        if key in self._cache:
             res = self._cache[key]
-        except KeyError:
```

Comment:
From what I understand, the `try` avoids a double lookup for cache hits. Or is there another reason for preferring `if` here?

Reply:
Centralizing the one way to fill the cache with valid data. 

I mean. You use your cache in 3 palaces. So it isn't so hard. But generally you want to define the function once and just forget about it

Reply:
i'm happy to add back the try except. let me do that now.